File: Shift.Infrastructure/Models/ViewModels/Auth/UndergraduateRegisterVM.cs
using System;
using Shift.DAL.Models.University;

namespace Shift.Infrastructure.Models.ViewModels.Auth
{
	public class UndergraduateRegisterVM
	{
		public UserRegisterVM User { get; set; }

		public EducationForm EducationForm { get; set; }

		public int? SpecialtyId { get; set; }

		public int AdviserId { get; set; }

		public DateTime? StartEducationDate { get; set; }
		public DateTime? FinishEducationDate { get; set; }
	}
}

File: Shift.Web/ClientApp/src/app/infrastracture/entities/auth/UndergraduateRegisterVM.ts
import { EducationForm } from "./EducationForm";
import { UserRegisterVM } from "./UserRegisterVM";

export class UndergraduateRegisterVM {
    public User: UserRegisterVM;

    public  EducationForm: number;

    public SpecialtyId: number;
    public AdviserId: number;

    public StartEducationDate: Date;
    public FinishEducationDate: Date;
}